libbeat/esleg/eslegclient/connection.go

```diff
@@ -311,16 +314,113 @@ func (conn *Connection) Close() error {
 	return nil
 }
 
+type httpClientProxySettings struct {
+	*httpcommon.HTTPClientProxySettings
+}
```

Comment:
Here you can just create a new type instead of embedding the struct:
```suggestion
type httpClientProxySettings httpcommon.HTTPClientProxySettings
```